### PR TITLE
Assorted UI updates

### LINKED ISSUE #12: Improve websqrl ui tweet view
When a Tweet is logged, or perhaps a create a new statement to output an event, the data presented on the right pane should be separated from other messages.

It's up to you how they're added and how the scrolling works. Right now though it seems like a jumble of log messages rather than individual tweet events.

### LINKED ISSUE #13: Improve websqrl ui on compilation failure
If the compile fails the error message should persist in the right pane (and perhaps show when the compile is running too)

Right now it's just logged above other messages, and so it's hard to see if the tweets are scrolling through. Compilation failure is expected while you're typing in new code, and the old code should keep running if the new code does not compile.

### PR DESCRIPTION
**Pull request by meyer**
# Problem

WebSQRL UI needs some love.

# Solution

I made a variety of changes:

- **dark mode support**. We can add a manual toggle too if we want.
- **small screen support**. The editor flips from row to column on narrow screens.
- **error marker support** in the editor. SQRL compile errors will now have their location highlighted once compilation is done. This closes #13.
- **minor tweet formatting updates**. The general layout is the same but I updated spacing and colours. Tweets are also now logged reverse-chronologically to match the Twitter UI. This closes #12 (though there is obviously more work to do).
  - One open question: should we load avatars in the tweet view?

# Result

Screenshot of the updated SQRL UI

Apologies for the chunky revision. Future revisions should be a bit more succinct and focused as we eliminate the low-hanging fruit.

### CODE REVIEW

File: tsconfig.base.json [hunk 1]
```diff
@@ -13,15 +13,14 @@
     "strict": true,
     "outDir": "./lib",
     "preserveConstEnums": true,
-    "removeComments": true,
-    "inlineSourceMap": false,
-    "typeRoots": ["./node_modules/@types"],
```
Comment (meyer) [inline tsconfig.base.json:18]: this is the default setting

File: websqrl/components/WikipediaEventStory.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  "TweetId",
+  "TweetDate",
+] as const;
+export type WikipediaFetchFeature = typeof WIKIPEDIA_FETCH_FEATURES[number];
+
+type WikipediaEventStoryProps = Omit<Result<WikipediaFetchFeature>, "type">;
+
+export const WikipediaEventStory: React.FC<WikipediaEventStoryProps> = () => {
+  return <Block>hello!</Block>;
+};
```
Comment (meyer) [inline websqrl/components/WikipediaEventStory.tsx:16]: this is a placeholder at the moment. The wikipedia demo is broken as of a few commits ago. I briefly tried to fix it but there's some web worker funkiness going on.

File: websqrl/package.json [hunk 3]
```diff
@@ -9,6 +9,7 @@
   },
   "dependencies": {
     "date-fns": "^2.29.3",
+    "jsxstyle": "canary",
```
Comment (meyer) [inline websqrl/package.json:12]: trying out some new experimental stuff

File: websqrl/tsconfig.worker.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{
+  "extends": "../tsconfig.json",
+  "include": ["**/*.worker.ts"],
+  "compilerOptions": {
+    "lib": ["WebWorker"]
```
Comment (meyer) [inline websqrl/tsconfig.worker.json:5]: this keeps us from having to do `any` shenanigans when trying to use `self`

File: websqrl/components/WikipediaEventStory.tsx (hunk 2 above)
Comment (qix) [reply websqrl/components/WikipediaEventStory.tsx:16]: Yeah, there's a lot of pieces missing of that demo. I'm treating it as dead for now, but not too hard to revive.

File: websqrl/src/MonacoEditor.tsx [hunk 5]
```diff
@@ -32,6 +34,21 @@ export const MonacoEditor: React.FC<MonacoEditorProps> = ({
...
+  useEffect(() => {
+    if (!editorRef.current || monacoEditorObj.state !== "success") return;
+    if (typeof navigator === "undefined") return;
+
+    const model = editorRef.current.getModel();
+    if (!model) return;
+
+    monacoEditorObj.value.editor.setModelMarkers(
+      model,
+      // i'm not sure what the owner should be
```
Comment (qix) [inline websqrl/src/MonacoEditor.tsx:46]: This comment might be lost, should we resolve or remove it?
(or explain a little more)

File: websqrl/src/constants.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+  boxShadowColor: "#333",
+  pageForeground: "#000",
+  secondary: "#888",
+  pageBackground: "#EEE",
+  insetBackground: "#FFF",
+  containerOutlineColor: "rgba(0,0,0,0.15)",
+  containerRadius: "6px",
+})
+  .addVariant("darkMode", {
+    mediaQuery: "screen and (prefers-color-scheme: dark)",
```
Comment (qix) [inline websqrl/src/constants.ts:15]: Fancy

**Review by qix — APPROVED** [review]
Some minor nits, but would love to see this merged

File: websqrl/src/MonacoEditor.tsx (hunk 5 above)
Comment (meyer) [reply websqrl/src/MonacoEditor.tsx]: Looks like it’s an ID that monaco uses to namespace markers. If you set markers using that ID, it’ll overwrite existing ones with that ID. The `removeAllMarkers` also takes the ID as its only param.

**Pull request merged**